Dockerfile

```diff
@@ -1,23 +1,6 @@
-FROM ubuntu:xenial as builder
-
-WORKDIR /build
-
-# install tools and dependencies
-RUN apt -y update && \
-	apt install -y --no-install-recommends \
-	software-properties-common curl git file binutils binutils-dev snapcraft \
-	make cmake ca-certificates g++ zip dpkg-dev python rhash rpm openssl gettext\
-	build-essential pkg-config libssl-dev libudev-dev ruby-dev time clang
-
-# install rustup
-RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
-
-# rustup directory
-ENV PATH /root/.cargo/bin:$PATH
-
-# setup rust beta and nightly channel's
-RUN rustup install nightly
-RUN rustup install stable
+# the WASM build of the runtime is completely indepedent 
+# we can avoid cache invalidations by running it in an extra container
+FROM rustlang/rust:nightly as wasm_builder
 
 # install wasm toolchain for polkadot
 RUN rustup target add wasm32-unknown-unknown --toolchain nightly
```

Comment:
instead of that, you could just call the init script?

Reply:
absolutely, but was like this and did not see a reason to change